worker/environ/environ.go

```diff
@@ -101,14 +113,30 @@ func (t *Tracker) loop() error {
 			if !ok {
 				return errors.New("environ config watch closed")
 			}
-		}
-		logger.Debugf("reloading environ config")
-		modelConfig, err := t.config.Observer.ModelConfig()
-		if err != nil {
-			return errors.Annotate(err, "cannot read environ config")
-		}
-		if err = t.environ.SetConfig(modelConfig); err != nil {
-			return errors.Annotate(err, "cannot update environ config")
+			logger.Debugf("reloading environ config")
+			modelConfig, err := t.config.Observer.ModelConfig()
+			if err != nil {
+				return errors.Annotate(err, "cannot read environ config")
+			}
+			if err = t.environ.SetConfig(modelConfig); err != nil {
+				return errors.Annotate(err, "cannot update environ config")
+			}
+		case _, ok := <-cloudWatcher.Changes():
+			if !ok {
```

Comment:
here you could do:
cloudSpecChanger, ok := t.environ.(CloudSpecSetter)
if !ok {
 logger.Warningf("cloud type %X doesn't support dynamic changing of cloud spec", t.environ)
 continue
}

That seems better than having it be an unrealized no op
I might slightly prefer to have all the Environs implement the interface, but have *them* log a message that it isn't actually supported, as it makes it easier to grep for what clouds providers need updating.

Reply:
yup, fixed